### ISSUE TITLE
configure on FreeBSD does not handle configuration file location

### ISSUE THREAD

**Issue by fluca1978** [pgbackrest/pgbackrest#1363]
pgbackrest version 2.33 (release) on FreeBSD 12.2-release.
I was expecting that having binaries and libraries under `/usr/local` now by default, even configuration should be placed on `/usr/local/etc` instead of `/etc`.
However it seems to me this is not.

Example:

```
 % uname -a
FreeBSD miguel 12.2-RELEASE FreeBSD 12.2-RELEASE r366954 GENERIC  amd64

 % ./configure && gmake && sudo gmake install
...

% ls -sh /usr/local/etc/pgbackrest.conf
4 /usr/local/etc/pgbackrest.conf
```

With everything in place, the following does not work:

```
% sudo -u postgres pgbackrest --stanza miguel info 
stanza: miguel
    status: error (missing stanza path)
```

while it works if I copy/move the configuration file to the default location:

```
 % sudo mv /usr/local/etc/pgbackrest.conf /etc 

 % sudo -u postgres pgbackrest --stanza miguel info 
stanza: miguel
    status: ok
    cipher: none

    db (current)
        wal archive min/max (12): 00000001000000C200000021/00000001000000C200000029
...
```

Interestingly, a link does not work either:

```
 % sudo ln -s /usr/local/etc/pgbackrest.conf /etc/pgbackrest

 % sudo -u postgres pgbackrest --stanza miguel info   
ERROR: [041]: unable to open file '/etc/pgbackrest/pgbackrest.conf' for read: [20] Not a directory

 % ls -l /etc/pgbackrest 
lrwxr-xr-x  1 root  wheel  30 Apr 12 12:37 /etc/pgbackrest -> /usr/local/etc/pgbackrest.conf
```

Configuring the software with `--with-configdir` works as expected:

```
% ./configure --with-configdir=/usr/local/etc && gmake && sudo gmake install
...
% ls /usr/local/etc/pgbackrest.conf
/usr/local/etc/pgbackrest.conf

 % ls -l /etc/pgbackrest.conf       
ls: /etc/pgbackrest.conf: No such file or directory

 % sudo -u postgres pgbackrest --stanza miguel info                          
stanza: miguel
    status: ok
    cipher: none

    db (current)
        wal archive min/max (12): 00000001000000C200000021/00000001000000C200000029

...
```

However, according to FreeBSD [hier(7)](https://www.freebsd.org/cgi/man.cgi?query=hier&apropos=0&sektion=7&manpath=FreeBSD+12.1-RELEASE+and+Ports&arch=default&format=html&sektion=7) , `/usr/local/etc` should mimic `/etc`, that I suspect means the configuration file should be placed there.
This is partially related to https://github.com/pgbackrest/pgbackrest/issues/1361#issuecomment-817630417

**Comment by dwsteele** [comment]
This sounds good but since none of us have FreeBSD experience hopefully @mschout or somebody commenting in #1293 will have a look.

**Comment by mschout** [comment]
So, to get the configs in `/usr/local/etc`, you need to use `--with-configdir`

When I made the pull request that added `--with-configdir`, I purposely did *NOT* use `/usr/local/etc` by default on FreeBSD everywhere because
- The config file location at the time was hard coded into pgbackrest
- I was terrified of breaking existing installs of pgbackrest on FreeBSD by other users who had compiled pgbackrest from sources since the config file *MUST* be at `/etc/pgbackrest.conf` previously.

So my thoughts at the time were along the lines of, we should not change the way "build it yourself from source" behaves by default, because I know with absolute certainty that there is an install base out there that built pgbackrest from source, on FreeBSD, and this would have broken the way things get installed for those users.  I felt if we were going to make such a drastic change such as changing the default location for the config file if you built the dist from source by yourself, at minimum, we need to warn users loudly in the changelog, and probably, something like a large version number bump.

However, the pending FreeBSD port, *does* set `--with-configdir` correctly so that the config file will end up in `/usr/local/etc` if you install it from ports. The thinking there was that if you switched from "build it myself from source code", to "install it from FreeBSD ports", you probably are more comfortable with the fact that you probably need to update the config file, since *everything* in ports goes under `/usr/local`.  

In other words, if I switched my own installation of pgbackrest from "built it myself from source" to "install from freebsd-ports", I would fully expect that I need to move the config file.   But if I just upgraded minor patchlevel of pgbackrest, and the default config file location changed, that would be pretty astonishing to me.  I wanted to avoid that.

So all of that said, if others within the `pgbackrest` community feel that its okay to make this change, I have no objection to it.

I don't pretend to have any sway on the decision making in the project :)  I leave that to people who have contributed far more than I :)

**Comment by dwsteele** [comment]
> So my thoughts at the time were along the lines of, we should not change the way "build it yourself from source" behaves by default, because I know with absolute certainty that there is an install base out there that built pgbackrest from source, on FreeBSD, and this would have broken the way things get installed for those users.

That's an excellent point, and I agree. We've definitely got complaints in the past when the behavior of `configure` has changed and I suppose this would be similar.

> I don't pretend to have any sway on the decision making in the project

Well, we won't make you responsible for the final decision, but certainly your input is valuable. It is simply not possible for the core team to have expertise on all platforms where pgBackRest may be deployed.

So these discussions are valuable, even if we end up leaving things as is, which seems to be where we are headed.

**Comment by mschout** [comment]
@dwsteele understood.  Happy to provide input that was the gist :)

**Comment by keithf4** [comment]
> So all of that said, if others within the `pgbackrest` community feel that its okay to make this change, I have no objection to it.
> 
> I don't pretend to have any sway on the decision making in the project :) I leave that to people who have contributed far more than I :)

That all sounds great to me. I would certainly expect the port the put things in the default locations expected for FreeBSD and if I wasn't doing things that way before, that's my own fault and need to get in line. :)

**Comment by fluca1978** [comment]
I clearly speak for myself, but so far I've always build pgbackrest from source and used the `/usr/local/etc/pgbackrest.conf` file on FreeBSD, as annoying as it is by means of specifying `--config` on the command line.
I'm still in doubt about the fact that a build from source should place the file in the "correct" place if the operating system can be detected correctly, I would like it to do but it is a personal opinion.

I think that the pgbackrest help screen should display always the path to the configuration file, in order to help users find out where it is. This clearly is not going to fix the break once a version moves a configuration file from one place to another.
I'm not sure too if we can emit a warning in the case there is a file in a "well know path" and `--with-configdir` is used (i.e., I want to compile using `/usr/local7etc/pgbackrest.conf` but a previous version created a `/etc/pgbackrest.conf`).

So far, I think we can close the issue.

**Comment by dwsteele** [comment]
> I think that the pgbackrest help screen should display always the path to the configuration file

It does:
```
$ pgbackrest help backup
<...>
  --config                       pgBackRest configuration file
                                 [default=/etc/pgbackrest/pgbackrest.conf]
<...>
```

**Comment by fluca1978** [comment]
> > I think that the pgbackrest help screen should display always the path to the configuration file
> 
> It does:
> 
> ```
> $ pgbackrest help backup
> <...>
>   --config                       pgBackRest configuration file
>                                  [default=/etc/pgbackrest/pgbackrest.conf]
> <...>
> ```

I meant displaying the configuration file on the main help screen. Something like:

```
$ pgbackrest help
pgBackRest 2.33 - General help

Usage:
    pgbackrest [options] [command]

Default configuration file (use --config-path to override):
   /usr/local/etc/pgbackrest.conf

Commands:
    archive-get     Get a WAL segment from the archive.
    archive-push    Push a WAL segment to the archive.
    backup          Backup a database cluster.
...
```

I'm wondering if this can help users to understand where to look on a fresh install.

**Comment by dwsteele** [comment]
Hmmm, maybe. But there are four config settings, so it may still be confusing.

It's fine if somebody wants to contribute it, but this doesn't come up enough for me to think it is an issue since we do make the data available in command help.

**Comment by fluca1978** [comment]
Closing this issue.
